### ISSUE TITLE
Training Size is truncated

### ISSUE THREAD

**Issue by vslobody** [keras-team/keras-tuner#429]
I have created tuner:
```
    tuner = ks.tuners.Hyperband(
        bm,
        objective='val_loss',
        max_epochs = 1,
        directory = 'options_dir',
        project_name = 'intro_to_hyperopt6',                         
        )
```
And when i try to perform search, looks like training only goes for the first 30 inputs:    
```
tuner.search(**train_generator**, epochs=1, validation_data = val_generator, steps_per_epoch=None)
  1/30 [>.............................] - ETA: 0s - loss: 0.4686 - mae: 0.5469 - mape: 2067.770
  9/30 [========>.....................] - ETA: 0s - loss: 0.7718 - mae: 0.6734 - mape: 3252.6553
17/30 [================>.............] - ETA: 0s - loss: 0.5454 - mae: 0.5530 - mape: 2823.6921
24/30 [=======================>......] - ETA: 0s - loss: 0.4311 - mae: 0.4815 - mape: 2528.2295
30/30 [==============================] - 2s 66ms/step - loss: 0.3705 - mae: 0.4402 - mape: 2325.3970 - val_loss: 0.9716 - val_mae: 0.7853 - val_mape: 2185.9128
[Trial complete]
```
However, the actual size of the train_generator is
```
**train_generator**.data.shape
(3878, 256)
```
with batch size 8 i should have like 469 iterations, not 30.

And i have the same problem when training using `model.fit`.

However, when i create supposedly the same model outside of the tuner.search loop, i have "normal" `model.fit` behavior:
`12/469 [..............................] - ETA: 6s - loss: 0.6506 - mae: 0.6444 - mape: 22295.4531
`
What could be a miss?

**Comment by vslobody** [comment]
OBE
